src/sagemaker_server.cc

```diff
@@ -764,11 +793,11 @@ SagemakerAPIServer::SageMakerMMEUnloadModel(
     evhtp_send_reply(req, EVHTP_RES_BADREQ);
     LOG_ERROR << "Unable to unregister model repository for path: "
               << repo_parent_path << std::endl;
-    TRITONSERVER_ErrorDelete(unload_err);
   } else {
     evhtp_send_reply(req, EVHTP_RES_OK);
   }
 
+  TRITONSERVER_ErrorDelete(unload_err);
```

Comment:
I think this is a segfault waiting to happen, since you're returning `nullptr` at the end of the `TRITONSERVER_ServerUnregisterModelRepository`.

`TRITONSERVER_ErrorDelete` does unconditional `delete` on the argument: https://github.com/triton-inference-server/core/blob/main/src/tritonserver.cc#L712-L717

Reply:
I think delete on `nullptr` will not do any harm, but I see the `unload_err` is reused for multiple API calls and there is chance of memory leak, i.e. `unload_err` may be set in line 743 and the pointer is overwritten in line 779

Reply:
You're right, sorry. delete on nullptr is actually fine. From https://en.cppreference.com/w/cpp/language/delete:

> If expression evaluates to a null pointer value, no destructors are called, and the deallocation function may or may not be called (it's unspecified), but the default deallocation functions are guaranteed to do nothing when passed a null pointer.

Reply:
Updated the code to not re-use the unload_err pointer